docs/developer/release/README.md

```diff
@@ -0,0 +1,22 @@
+# Release instructions
+
+The documents in this directory are targeted towards core Dangerzone developers
```

Comment:
Right now, the core devs are doing the releasing, but I believe it's actually a different role

```suggestion
The documents in this directory are targeted towards Dangerzone release managers
```